### ISSUE TITLE
build with docker, gets some errors

### ISSUE THREAD

**Issue by kylongmu** [OpenNuvoton/MA35D1_Docker_Script#1]
Use docker to build openwrt22 get this error:
run join.sh , pull source code of openwrt, after execute "./scripts/feeds install -a".
gets some errors:
`
xargs: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by xargs)
find: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by find)
find: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by find)
sed: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by sed)
sed: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by sed)
xargs: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by xargs)
`
--
Build yocto is ok!

**Comment by ccchang12** [comment]
@kylongmu Please run command "ldd /usr/bin/xargs" in your docker environment, and provide the result.

**Comment by kylongmu** [comment]
> @kylongmu Please run command "ldd /usr/bin/xargs" in your docker environment, and provide the result.

user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ ldd /usr/bin/xargs
	linux-vdso.so.1 (0x00007ffd8ffc1000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3b54c47000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f3b54e5a000)

**Comment by ccchang12** [comment]
What is your host platform for docker? Such as Ubuntu 20.04 @ Virtual-box.

**Comment by kylongmu** [comment]
> 



> What is your host platform for docker? Such as Ubuntu 20.04 @ Virtual-box.

https://www.nuvoton.com/resource-download.jsp?tp_GUID=SW182022101516122042

**Comment by ccchang12** [comment]
We cannot duplicate your issue. Please runs below commands to check the GLIBC version and xargs is workable in docker.

cczhang@1d6613ab52ce:~/shared/docker_openwrt/Nuvoton-OpenWrt-22.03$ ldd --version
ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.

cczhang@1d6613ab52ce:~/shared/docker_openwrt/Nuvoton-OpenWrt-22.03$ ls /usr/bin/cat | xargs ls -l
-rwxr-xr-x 1 root root 43416 Sep  5  2019 /usr/bin/cat

Please provide the H/W and S/W detail information of your host machine, such as Intel i5-10400 with Virtual-box@Windows 10.  And can you do the same test on another machine?

**Comment by kylongmu** [comment]
> We cannot duplicate your issue. Please runs below commands to check the GLIBC version and xargs is workable in docker.
> 
> cczhang@1d6613ab52ce:~/shared/docker_openwrt/Nuvoton-OpenWrt-22.03$ ldd --version ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31 Copyright (C) 2020 Free Software Foundation, Inc. This is free software; see the source for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. Written by Roland McGrath and Ulrich Drepper.
> 
> cczhang@1d6613ab52ce:~/shared/docker_openwrt/Nuvoton-OpenWrt-22.03$ ls /usr/bin/cat | xargs ls -l -rwxr-xr-x 1 root root 43416 Sep 5 2019 /usr/bin/cat
> 
> Please provide the H/W and S/W detail information of your host machine, such as Intel i5-10400 with Virtual-box@Windows 10. And can you do the same test on another machine?


12th Gen Intel(R) Core(TM) i9-12900H notebook win11 enterprise

`user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ ldd /usr/bin/xargs
	linux-vdso.so.1 (0x00007ffd8ffc1000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3b54c47000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f3b54e5a000)
user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ ldd --version
ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ ls /usr/bin/cat | xargs ls -l
-rwxr-xr-x 1 root root 43416 Sep  5  2019 /usr/bin/cat
user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ 
`

**Comment by ychuang3** [comment]
Please find the user manual:
https://www.nuvoton.com/export/resource-files/en-us--UM_EN_MA35D1_Yocto.pdf
It is suggested to follow the advise of section 2.1 and 2.2 to create the evelopment environment with docker.
Note that installation mentioned in section 2.2 should be proceed in docker.

**Comment by ccchang12** [comment]
@kylongmu Could you please do the test again. Only do following procedures, and have no any other procedure.
1. Use VMware to open the ZIP-extracted folder as new virtual machine again
2. Run "git clone git@github.com:OpenNuvoton/Nuvoton-OpenWrt-22.03.git" command to download OpenWrt source
3. Run "./scripts/feeds update -a" 和 "./scripts/feeds install -a" commands

**Comment by kylongmu** [comment]
> Please find the user manual: https://www.nuvoton.com/export/resource-files/en-us--UM_EN_MA35D1_Yocto.pdf It is suggested to follow the advise of section 2.1 and 2.2 to create the evelopment environment with docker. Note that installation mentioned in section 2.2 should be proceed in docker.

Yocto build is ok, this problem is on Nuvoton-OpenWrt-22.03

**Comment by kylongmu** [comment]
> @kylongmu Could you please do the test again. Only do following procedures, and have no any other procedure.
> 
> 1. Use VMware to open the ZIP-extracted folder as new virtual machine again
> 2. Run "git clone [git@github.com](mailto:git@github.com):OpenNuvoton/Nuvoton-OpenWrt-22.03.git" command to download OpenWrt source
> 3. Run "./scripts/feeds update -a" 和 "./scripts/feeds install -a" commands

user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ ./scripts/feeds update -a
Updating feed 'packages' from 'https://git.openwrt.org/feed/packages.git^426ccd2e0cf56394017d32b864cd66a570852706' ...
Updating feed 'luci' from 'https://git.openwrt.org/project/luci.git^487e58a0a61f8d2396893bd9fa6695269b1fd1d3' ...
Updating feed 'routing' from 'https://git.openwrt.org/feed/routing.git^8872359011eee64981804f7ed42d5d6b54add6d8' ...
Updating feed 'telephony' from 'https://git.openwrt.org/feed/telephony.git^1d2031a5c82816483c51bca15649e2957fbe2bc2' ...
Create index file './feeds/packages.index' 
find: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by find)
find: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by find)
xargs: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by xargs)
sed: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by sed)
sed: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by sed)

**Comment by kylongmu** [comment]
The docker is based on:
user@aa3c667d7ce6:~/shared/Nuvoton-OpenWrt-22.03$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.5 LTS
Release:	20.04
Codename:	focal

--------------------------
But I find the GLIBC_2.33 need 20.10 at least, or maybe the apt source do not have the correspond libc version for 20.04. 
And I build Nuvoton-OpenWrt-22.03 in the 22.04 host is ok, this problem is only appear in the docker .

**Comment by ccchang12** [comment]
@kylongmu We have tested this Ubuntu VMware image on two different machines (both are Windows 10), and everything is OK. The Ubuntu 20.04 has only up to GLIBC 2.31, but related commands do not need the GLIBC 2.33 or GLIBC 2.34. I think if you want to use this Ubuntu VMware image, you can try to upgrade the GLIBC version of Ubuntu 20.04.

**Comment by kylongmu** [comment]
> 

Yes, I upgrade the GLIBC version of Ubuntu 20.04, and the build success. 
The error caused by the docker env (required by xargs)(required by find)(required by sed)(required by sed)(required by xargs) ,
so I guess maybe is do "apt upgrade" in the docker container make the env been caos.
